### PR TITLE
[Feature] Support select into outfile without broker

### LINKED ISSUE #9200: [Feature] Support select xxx into outfile without broker

### PR DESCRIPTION
**Pull request by xiaoyong-z**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9200

## Problem Summary(Required) ：

Support select xxx into outfile without broker

### CODE REVIEW

File: be/src/fs/fs.cpp [hunk 1]
```diff
@@ -74,15 +75,15 @@ inline bool is_staros_uri(std::string_view uri) {
     return false;
 }
 
-StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::string_view uri) {
+StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::string_view uri, FSOptions options) {
```
Comment (chaoyli) [inline be/src/fs/fs.cpp:78]: const FSOptions&

File: be/src/fs/fs.cpp [hunk 2]
```diff
@@ -92,15 +93,15 @@ StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::st
     return Status::NotSupported(fmt::format("No FileSystem associated with {}", uri));
 }
 
-StatusOr<std::shared_ptr<FileSystem>> FileSystem::CreateSharedFromString(std::string_view uri) {
+StatusOr<std::shared_ptr<FileSystem>> FileSystem::CreateSharedFromString(std::string_view uri, FSOptions options) {
```
Comment (chaoyli) [inline be/src/fs/fs.cpp:96]: const FSOptions&

File: be/src/fs/fs_hdfs.cpp [hunk 3]
```diff
@@ -194,8 +269,100 @@ class HdfsFileSystem : public FileSystem {
...
+
+    FSOptions _options;
 };
 
+Status HdfsFileSystem::path_exists(const std::string& path) {
+    std::string namenode;
+    RETURN_IF_ERROR(get_namenode_from_path(path, &namenode));
+    HdfsFsHandle handle;
+    RETURN_IF_ERROR(HdfsFsCache::instance()->get_connection(namenode, &handle, _options));
+    if (handle.type != HdfsFsHandle::Type::HDFS) {
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:284]: add path in the error status

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 23 (34.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|22|36.36%|[165, 166, 188, 224, 225, 226, 227, 229, 230, 231, 232, 233, 234, 235]|

**Comment by xiaoyong-z** [comment]
run starrocks_be_build

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_clang-format

**Comment by xiaoyong-z** [comment]
run starrocks_clang-tidy

**Comment by xiaoyong-z** [comment]
run starrocks_tscancode

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 23 (34.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|22|36.36%|[165, 166, 188, 224, 225, 226, 227, 229, 230, 231, 232, 233, 234, 235]|

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java [hunk 4]
```diff
@@ -169,8 +182,11 @@ private void getBrokerProperties(Set<String> processedPropKeys) {
                 processedPropKeys.add(entry.getKey());
             }
         }
-
-        brokerDesc = new BrokerDesc(brokerName, brokerProps);
+        if (!outfile_without_broker) {
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:185]: outfile_without_broker always false?

**Review by meegoo — CHANGES_REQUESTED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 24 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/ResultSink.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|8|23|34.78%|[165, 166, 168, 189, 225, 226, 227, 228, 230, 231, 232, 233, 234, 235, 236]|

**Pull request merged**